src/main/js/dynamic/loaded/users/PermissionTable.js

```diff
@@ -233,15 +234,27 @@ class PermissionTable extends Component {
         );
     }
 
-    onSavePermissions() {
+    async onSavePermissions() {
+        await this.setState({
```

Comment:
In 6.0.0 I may consider using promises here and refactor this code to use the .then function to control order.